### PR TITLE
Get authenticated using the token

### PR DESCRIPTION
**Pull request by Eleonore9**
Checks the token can be unsigned w/ the public key.
The unsign function checks the validity of the token.

### CODE REVIEW

File: src/kixi/himinbjorg/handler.clj [hunk 1]
```diff
@@ -1,11 +1,35 @@
...
+(defn get-web-service [req]
+  {:status 200 :body "Test web service - Himinbjorg"})
+
+(def public-key
+  (-> (try (slurp (io/file (System/getProperty "user.home")
+                           ".himinbjorg-conf.edn"))
+           (catch java.io.FileNotFoundException _
+             (slurp (io/resource "himinbjorg-test-conf.edn"))))
+      edn/read-string
+      :pubkey))
```
Comment (otfrom) [inline src/kixi/himinbjorg/handler.clj]:
```
  (defn file-exists? [f]
    (when (.exists (io/file f))
      f))
  
  (def public-key
    (-> (if-let [config (file-exists?
                         (io/file (System/getProperty "user.home")
                                  ".himinbjorg-conf.edn"))]
          (slurp config)
          (slurp (io/resource "himinbjorg-test-conf.edn")))
        edn/read-string
        :pubkey))
```

I can't decide if I like that more than your exception based code above. I generally don't like using exceptions where things aren't... exceptional. wdyt?

**Comment by otfrom** [comment]
a few questions

File: src/kixi/himinbjorg/handler.clj (hunk 1 above)
Comment (Eleonore9) [inline src/kixi/himinbjorg/handler.clj]: I like that, too. I think I'll let Elise decide what's best in this case.

File: src/kixi/himinbjorg/handler.clj [hunk 2]
```diff
@@ -1,11 +1,35 @@
...
+
+(def auth-backend
+  (jws-backend {:secret (ks/public-key
+                         (try (io/resource public-key)
+                              (catch java.io.FileNotFoundException _
+                                (io/file public-key))))
+                :token-name "Token"}))
```
Comment (Eleonore9) [inline src/kixi/himinbjorg/handler.clj]: CircleCI uses the config file in resources (and a test key) while locally we use a config in the home dir that points to another key.

File: src/kixi/himinbjorg/handler.clj (hunk 1 above)
Comment (elisehuard) [inline src/kixi/himinbjorg/handler.clj]: Sounds good indeed, we will have to do that elsewhere.

File: src/kixi/himinbjorg/handler.clj (hunk 2 above)
Comment (elisehuard) [inline src/kixi/himinbjorg/handler.clj]: I guess (thinking about it again) we could use different keys for either resource or absolute paths, maybe that would be neater

**Comment by otfrom** [comment]
:shipit:

**Pull request merged**